app/src/main/java/com/neura/sampleapplication/NeuraManager.java

```diff
@@ -44,14 +63,160 @@ public NeuraApiClient getClient() {
+        AnonymousAuthenticationRequest request = new AnonymousAuthenticationRequest(pushToken);
+
+        //Pass the AnonymousAuthenticationRequest instance and register a call back for success and failure events.
+        NeuraManager.getInstance().getClient().authenticate(request, new AnonymousAuthenticateCallBack() {
+            @Override
+            public void onSuccess(AnonymousAuthenticateData data) {
+                NeuraManager.getInstance().getClient().registerAuthStateListener(silentStateListener);
+                Log.i(TAG, "Successfully requested authentication with neura. ");
+                context.setUIState(true, true);
+            }
+
+            @Override
+            public void onFailure(int errorCode) {
+                NeuraManager.getInstance().getClient().unregisterAuthStateListener();
+                Log.e(TAG, "Failed to authenticate with neura. " + "Reason : " + SDKUtils.errorCodeToString(errorCode));
+                context.loadProgress(false);
+                context.mRequestPermissions.setEnabled(true);
+            }
+        });
+    }
+
+
+    /**
+     * Authenticate with Neura
+     * Receiving unique neuraUserId and accessToken (for external api calls : https://dev.theneura.com/docs/api/insights)
+     */
+    public static void authenticateByPhone(String phoneNumber, final FragmentMain context) {
+        AuthenticationRequest request = new AuthenticationRequest();
+        request.setPhone(phoneNumber);
+        NeuraManager.getInstance().getClient().authenticate(request, new AuthenticateCallback() {
+            @Override
+            public void onSuccess(AuthenticateData authenticateData) {
+                Log.i(getClass().getSimpleName(), "Successfully authenticate with neura. NeuraUserId = "
+                        + authenticateData.getNeuraUserId() + ". AccessToken = " + authenticateData.getAccessToken());
+                context.setUIState(true, true);
+
+                /**
+                 * Go to our push notification guide for more info on how to register receiving
+                 * events via firebase https://dev.theneura.com/docs/guide/android/pushnotification.
+                 * If you're receiving a 'Token already exists error',make sure you've initiated a
+                 * Firebase instance like {@link com.neura.sampleapplication.activities.MainActivity#onCreate(Bundle)}
+                 * http://stackoverflow.com/a/38945375/5130239
+                 */
+                NeuraManager.getInstance().getClient().registerFirebaseToken(context.getActivity(),
+                        FirebaseInstanceId.getInstance().getToken());
+
+                //TODO put here a list of events that you wish to receive. Beware, that these events must be listed to your application on our dev site. https://dev.theneura.com/console/apps
+                List<String> events = Arrays.asList("userArrivedHome", "userArrivedHomeFromWork",
```

Comment:
@Jfleit Please reduce the size of this list. leave just arrive and left home and work, and start finish driving, walking ,running.

Reply:
I can do that. Another option, which to me seems preferable, is to use AuthenticateData.getEvents() to receive the list of events registered by the developer in the Neura Console. Let me know which I should do.

Reply:
AuthenticateData.getEvents is only relevant in phone authentication where authenticatedata object returns in the callback.
But I'd rather not, it's a bad practice to register to *all* events